### PR TITLE
Refactor Model Reader on the Stella example

### PR DESCRIPTION
**Pull request by wkerzendorf**
this is a prototype implementation

@jvshields

### CODE REVIEW

File: tardis/io/model/tests/test_model_io.py
Comment (jvshields) [inline tardis/io/model/tests/test_model_io.py:1]: I'm a little confused about the test here. Is this going to test a specific file? And if not, do all stella models have these specific inner boundary and total masses?

**Review by jvshields — APPROVED** [review]

File: docs/io/model/index.rst [hunk 1]
```diff
@@ -0,0 +1,10 @@
+**************
+Reading Models
+**************
+
+TARDIS can read a variety of models. The following models are currently supported:
+
+.. toctree::
+    :maxdepth: 1
```
Comment (andrewfullard) [inline docs/io/model/index.rst:8]: Empty list?

File: tardis/io/model/stella.py
[the remark this answers is not in the document]
Comment (andrewfullard) [reply tardis/io/model/stella.py:1]: Agreed

File: tardis/io/model/stella.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            if i == len(header_re) - 1:
+                break
+        header["t_max"] = float(header["t_max"]) * u.day
+        header["zones"] = int(header["zones"])
+        header["inner_boundary_mass"] = (
+            float(header["inner_boundary_mass"]) * u.g
+        )
+        header["total_mass"] = float(header["total_mass"]) * u.g
+
+    data = pd.read_csv(fname, delim_whitespace=True, skiprows=7)
```
Comment (andrewfullard) [inline tardis/io/model/stella.py:54]: Consider defining this magic number and using it as the length of the loop.

File: tardis/io/model/tests/test_model_io.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import numpy as np
+
+from astropy import units as u
+
+from pathlib import Path
+
+from tardis.io.model import read_stella_model
+
+MODEL_DATA_PATH = Path(__file__).parent / "data"
```
Comment (andrewfullard) [inline tardis/io/model/tests/test_model_io.py:9]: Not a huge fan of using __file__, why not just `..`? Or is this more platform agnostic?

File: tardis/io/model/stella.py (hunk 2 above)
Comment (wkerzendorf) [reply tardis/io/model/stella.py:54]: happy to define this

File: tardis/io/model/tests/test_model_io.py (hunk 3 above)
Comment (wkerzendorf) [reply tardis/io/model/tests/test_model_io.py:9]: there is no way of guaranteeing in what directory you are when the tests run. This gets you definitely the location of this file absolutely.

File: tardis/io/model/stella.py
[the remark this answers is not in the document]
Comment (wkerzendorf) [reply tardis/io/model/stella.py:1]: I disagree with some of that. I believe in this case the metadata is contained in the header and so I think for the parsing using `HEADER`. but the dictionary that is named `header` should be renamed - I agree.

File: tardis/io/model/stella.py (hunk 2 above)
Comment (wkerzendorf) [reply tardis/io/model/stella.py]: done. I can't use it in the loop though as there are empty rows.

**Review by andrewfullard — APPROVED** [review]

**Review by jvshields — APPROVED** [review]

**Comment by andrewfullard** [comment]
Need Black formatting pass!

File: tardis/io/model/tests/test_model_io.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import numpy as np
+
+from astropy import units as u
+
+from pathlib import Path
+
+from tardis.io.model import read_stella_model
```
Comment (jvshields) [inline tardis/io/model/tests/test_model_io.py:7]: Does this import actually work? Should it be from tardis.io.model.stella?

Comment (jvshields) [reply tardis/io/model/tests/test_model_io.py:7]: Oh nevermind I see. You added the read_stella_model to __init__.py

**Review by jvshields — DISMISSED** [review]

**Review by andrewfullard — DISMISSED** [review]

**Review by jvshields — APPROVED** [review]
Any idea why the black workflow is failing?

**Review by andrewfullard — APPROVED** [review]

File: tardis/io/model/stella.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
+import re
+import dataclasses
+
+import pandas as pd
+from astropy import units as u
+
+
```
Comment (wkerzendorf) [inline tardis/io/model/stella.py:7]:
```suggestion
# The dataclass that holds the model in the end.
```

File: tardis/io/model/tests/test_model_io.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        stella_model_example_file1.metadata["t_max"].to(u.day).value, 50.0
+    )
+    np.testing.assert_almost_equal(
+        stella_model_example_file1.metadata["inner_boundary_mass"]
+        .to(u.g)
+        .value,
+        5.190242521200000e33,
+    )
+    np.testing.assert_almost_equal(
+        stella_model_example_file1.metadata["total_mass"].to(u.g).value,
```
Comment (wkerzendorf) [inline tardis/io/model/tests/test_model_io.py:33]:
```suggestion
   TOTAL_MASS =  2.618867335600000e34   stella_model_example_file1.metadata["total_mass"].to(u.g).value,
```

File: tardis/io/model/tests/test_model_io.py [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+        2.618867335600000e34,
```
Comment (wkerzendorf) [inline tardis/io/model/tests/test_model_io.py:34]:
```suggestion
        TOTAL_MASS,
```

**Review by jvshields — APPROVED** [review]

**Pull request merged**